test_util/cluster.py

```diff
@@ -184,6 +186,90 @@ def check_ssh(self, hosts):
         """
         self.ssher.remote_cmd(hosts, ['echo'])
 
+    def mesos_metrics_snapshot(self, host):
+        """Return a snapshot of the Mesos metrics for host."""
+        if host in self.masters:
+            port = 5050
+        else:
+            port = 5051
+
+        with self.ssher.tunnel(host) as tunnel:
+            return json.loads(
+                tunnel.remote_cmd([
+                    'curl', '--silent', '{}:{}/metrics/snapshot'.format(host.private_ip, port)
+                ]).decode('utf-8')
+            )
+
+    def zk_mode(self, host):
+        """Return the mode of the ZooKeeper instance on host."""
+        with self.ssher.tunnel(host) as tunnel:
+            stat_out = tunnel.remote_cmd([
+                'sudo', 'docker', 'run',
```

Comment:
Kazoo would be ideal, but I'm deliberately SSHing to the host and then talking to the client port with the expectation that this method will be easier to maintain as masters get more secure. ZooKeeper docs recommend netcat, they don't appear to provide a tool for sending 4-letter commands. I'll update this to use the netcat bin we already have installed.